spec/betterc.dd

```diff
@@ -7,6 +7,11 @@ $(HEADERNAV_TOC)
 $(P BetterC is a subset of D that doesn't depend on the D runtime library, only the C runtime library.
 )
 
+$(P $(RED Warning:) There may be hidden assumptions built into a D compiler and its matching runtime.
+As an advanced compiler feature, there may be aspects to it that have not been accounted for i.e. bindings in druntime that have C macros translated as non-templated free-functions.
```

Comment:
Well I fixed that specific thing :slightly_smiling_face:  https://github.com/dlang/dmd/pull/21108

I think the phrasing 'there may be hidden assumptions' is too abstract for a warning targeted to new D users. I'd phrase it as "While the D runtime library and standard library are not linked in, it is still possible to import several `core.*` and `std.*` packages and use certain templates, types, and extern definitions from them. However, these libraries were not written with `-betterC` in mind so you are on your own with figuring out which functions you can and can't use. Expect cryptic compiler or linker errors when trying this out."

Reply:
I slightly rewrote it.

Oh and good job with that PR.

Reply:
Thanks. This warning is also a nice addition.